### PR TITLE
Create CRUD from post table

### PR DESCRIPTION
**Pull request by Zama15**
## Description

- What changes are being made?
were added the crud from the post table following MVC pattern and the architecture Restful API for the controller and the model
- Why are these changes necessary?
To create a standard pattern in the project which is common in the frameworks
- How have these changes been tested?
Using apache2 and postgreSQL in the localhost

## List of Changes

1. Create the DB (in db folder)
   - the db.sql contains the database structure
   - the procedures.sql contains functions to get data from the table (search, read) and procedures to manipulate the table (create, update, delete)
2. Create the model base (in models folder)
   - the database class which made the connection to the database
   - the post class inherit the method to create the connection with the database and execute the procedures or function in created returning an Array with the status of the execution(Boolean) and the data (Associative Array) if was a function or a message(String) if were a error
3. create the controllers (in controllers folder)
   - posts controller contains the restful standard API (create, edit, patch, delete) and the methods to redirect to the respective view in the views/posts folder
   - create the application controller, this has utilities functions used in multiple files in the commit, check the documentation inside the file for more information
4. Create views (in views folder)
   1. layouts
      - this folder is to use a partial used multiple times in views as main usage
      - header, sidebar main and sidebar chats were redeclared here
      - the ***application.php is the main file where the user just have access to*** the controller jsut manage which partial used base on the controller and the action the user is trying the access for example: `localhost/views/layouts/application.php?controller=posts&action=index` for more details check the documentation inside the controllers
    2. errors
       - 404 were the page cant be reach or not exist (used as if the post were not found)
       - 500 as server error (used as if the bd connection were not successfully creted)
    3. posts
       - this are the partials that are render when the user request an action from a controller
       - the ones created correspond to the standart get routes used in the frameworks (index, new, edit, show) to send the POST data to the controller in the restful API methods in the controller

## Additional Information

index action:
image
show action:
image
create action:
image
edit action:
image

## Checklist

- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [ ] New and existing unit tests pass locally with my changes.

### CODE REVIEW

**Review by juperameza — APPROVED** [review]
LGTM

**Review by Gavalencia12 — APPROVED** [review]

**Comment by IsmaelMontiel710** [comment]
You can make it so that when something is published, images are also added?

**Comment by Zama15** [comment]
> You can make it so that when something is published, images are also added?

i also think about that but in the state of how table is rn, i am not pretty sure if is the best. the post just can be able to save one image per table, and i thought on leave the option to the client to attatch multiple images, but with that the code it will be so much and even with the atomic commits it will be a lot to contain in a single PR, i motivate you to make a issue and asign to me to finish immediately after this merge

**Review by AngelDiaz591 — APPROVED** [review]

File: db/PostgreSQL/db.sql
Comment (IsmaelMontiel710) [inline db/PostgreSQL/db.sql:1]: is mysql, please pay attencion

**Review by IsmaelMontiel710 — COMMENTED** [review]
excellent work, congratulations

File: db/PostgreSQL/db.sql
Comment (Zama15) [reply db/PostgreSQL/db.sql:1]: yep, next commit mine i will change that

**Pull request merged**